src/coreclr/inc/cordebuginfo.h

```diff
@@ -431,4 +431,31 @@ class ICorDebugInfo
         // Source information about the IL instruction in the inlinee
         SourceTypes Source;
     };
+
+    struct AsyncContinuationVarInfo
+    {
+        // IL number of variable (or one of the special IL numbers, like UNKNOWN_ILNUM)
+        uint32_t VarNumber;
+        // Offset in continuation's data where this variable is stored
+        uint32_t Offset;
+    };
+
+    struct AsyncSuspensionPoint
+    {
+        // IL offset in the root method that resulted in the creation of this suspension point.
+        uint32_t RootILOffset;
+        // Index of inline tree node containing the IL offset (0 for root)
+        uint32_t Inlinee;
+        // IL offset that resulted in the creation of the suspension point.
+        uint32_t ILOffset;
```

Comment:
if we care about size, can IL offsets inside a method be `uint16_t` ?

Reply:
I don't think there is a 2^16 size limit on IL, but in any case we do not care about size in these data types. The actual storage of these is compressed in the VM, so when these values are small, they consume little space.

Reply:
Right. IL size is 32 bit, and branches take 4 byte offsets, unless they are `.s` flavors. 
Do not believe AI if it says otherwise :-)